service/history/replicationTaskProcessor.go

```diff
@@ -269,6 +270,91 @@ func (p *ReplicationTaskProcessor) processTaskOnce(replicationTask *r.Replicatio
 	return err
 }
 
+func (p *ReplicationTaskProcessor) putReplicationTaskToDLQ(replicationTask *r.ReplicationTask) error {
+	request, err := p.generateDLQRequest(replicationTask)
+	if err != nil {
+		p.logger.Error("Failed to generate DLQ replication task.", tag.Error(err))
+		// We cannot deserialize the task. Dropping it.
+		return nil
```

Comment:
why dropping a task?

Reply:
DLQ has the same schema as replication task. So if we cannot deserialize the task then there is not much we can do. We can potentially create a new blob column to store the payload. Not sure if it's worth it.